src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -47,7 +47,7 @@ class ReservedSpace {
 
   // ReservedSpace
   ReservedSpace(char* base, size_t size, size_t alignment,
-                size_t page_size, bool special, bool executable);
+                size_t page_size, bool special, bool executable, MemTag mem_tag);
```

Comment:
The growth in complexity is unfortunate.

Reply:
I think the cost is worth the fact that we mark the memory with a tag as soon as possible, so that we don't catch NMT with its trousers down.